documentation/src/main/asciidoc/topics/query.adoc

```diff
@@ -18,480 +18,570 @@ Apart from indexed queries, {brandname} can run queries over non-indexed or
 partially indexed data.
 
 In terms of Search APIs, {brandname} has its own query language called _link:#query_ickle[Ickle]_, which is string-based and adds support for full-text
-querying. The link:#query_dsl[Query DSL] can be used for both embedded and remote java clients when full-text is not required; for Java embedded clients
-{brandname} offers the link:#query_hibernatesearch[Hibernate Search Query API] which supports running Lucene queries in the grid, apart from advanced search capabilities
-like Faceted and Spatial search.
+querying.
 
 Finally, {brandname} has support for link:#query_continuous[Continuous Queries], which works in a reverse manner to the other APIs: instead of creating, executing a query
 and obtain results, it allows a client to register queries that will be evaluated continuously as data in the cluster changes, generating notifications
 whenever the changed data matches the queries.
 
-[[query_library]]
-== Embedded Querying
+== Indexing
```

Comment:
Creating Indexes for {brandname} Caches? | Indexing Entries in {brandname} Caches

When we say indexing here, what do we really mean? Is it an index of the cache or an index of the entries in the cache? Maybe it's just me but I think expanding the title out helps with understanding.

Reply:
That's a good point. We should document that indexing is on the cache values.